packages/contracts/contracts/SmartInvoice.sol

```diff
@@ -130,6 +139,43 @@ contract SmartInvoice is
         emit Register(_client, _provider, amounts);
     }
 
+    function extendTerminationTime(uint256 _time) public {
+        require(!locked, "locked");
+        require(_msgSender() == client, "msg.sender !client");
+        terminationTime += _time;
+
+        emit TerminationExtension(_msgSender(), _time);
+    }
+
+    function addMilestones(uint256[] calldata _milestones, uint256 _time)
+        external
+    {
+        require(!locked, "locked");
+        require(_msgSender() == client, "msg.sender !client");
```

Comment:
The provider will want to be able to add milestones, I don't see this requirement as reasonable.

Reply:
will change